File: java/Ejercicio2.java
package excepciones;

public class Ejercicio2 {

	public static void main(String[] args) {
		
		//instancias
		GatoSimple garfield = new GatoSimple("macho");
		GatoSimple lisa = new GatoSimple("hembra");
		GatoSimple tom = new GatoSimple("macho");
		
		/*si se aparean gatos del distinto sexo, devuelve un objeto gato y del mismo salta excepcion*/
		try {
			System.out.println(tom.apareaCon(lisa));
		} catch (ExceptionApareamientoImposible e) {
			System.out.println(e.getMessage());
		}
		
	}

}

File: python/Ejercicio2.py
'''
Prueba del metodo apareaCon con excepciones

@author: Rafael Infante
'''

'''
Nota si la clase a importar esta fuera del paquete se importaria asi.
'''
from excepciones.GatoSimple import GatoSimple

garfield = GatoSimple("macho")
tom = GatoSimple("macho")
lisa = GatoSimple("hembra")

try:
    print(garfield.apareaCon(lisa))
except ValueError as e:
    print(e)


